test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs

```diff
@@ -18,6 +22,9 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests;
 
 public sealed class MockCollectorIntegrationTests
 {
+    private static int gRPCPort = 4317;
```

Comment:
Why use `static`? I think we could get rid of the `Interlocked.Increment` statements if we mark these integer as instance fields. Every `Fact` or `Theory` should create its own instance of `MockCollectorIntegrationTests`.

Reply:
We have this issue currently: https://github.com/open-telemetry/opentelemetry-dotnet/issues/4929. I am trying to see if this fixes the issue. We had a similar issue with asp.net core tests and having separate port helped fixing it.

Reply:
Okay interesting. In that case, `Interlocked.Increment` should ensure that none of tests are using the same port.